crates/apollo-compiler/src/database/hir.rs

```diff
@@ -1326,6 +1333,20 @@ impl Float {
             inner: OrderedFloat(float),
         }
     }
+
+    pub fn get(self) -> f64 {
+        self.inner.0
+    }
+
+    /// Convert the value to the nearest i32, if it is finite and in the i32 range.
```

Comment:
Does it round to the nearest i32 or does it truncate? I thought `as` truncates

Reply:
I should have checked rather than write this from memory :sweat_smile: https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast says float `as` int rounds towards zero. I’ll fix the doc-comment.

(This public API could be used in cases where this makes a difference, but it’s mostly useful when matching on `hir::Value::Int` where the conversion will either fail or be exact.)